app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -40,10 +43,14 @@ class WallpaperManager(
     val wallpapers = allWallpapers
         .filter(::filterExpiredRemoteWallpapers)
         .filter(::filterPromotionalWallpapers)
+        .also {
+            appStore.dispatch(AppAction.WallpaperAction.UpdateAvailableWallpapers(it))
+        }
```

Comment:
is this intended to be a first time initialization Action?

Reply:
Probably initially, yeah. I picture this either being run automatically by the `WallpaperUseCase` when it is initialized, or as a side-effect of the `WallpaperUseCase` being called from `FenixApplication::onCreate`. I can the imagine the timing of when it's dispatched changing based on longer-term feature discovery, but it's hard to see it happening more than once during the lifetime of the app.